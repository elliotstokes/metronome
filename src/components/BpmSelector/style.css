.bpm-button {
  border-radius: 5px;
  background: rgb(45, 28, 63);
  border-radius: 5px;
  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.5);
  letter-spacing: 0.78px;
  color: rgb(142, 172, 255);
  font-size: 14px;
  border: 0;
  margin: 5px;
  width: 86px;
  height: 35px;
}

.bpm-selected {
  color: rgb(22, 16, 32);
  background:rgb(142, 172, 255)
}

.bpm-selector {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  max-width: 400px;
  margin-bottom: 30px;
}

.bpm-selector>* {
  flex: 0 0 33.3333%;
}