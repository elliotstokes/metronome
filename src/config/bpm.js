import bpmData from '../resources/data/bpm.json'

export const bpmInfo = bpmData.reduce((info, current) => {
  const [title, artist, bpm] = current
  if (info[bpm]) {
    info[bpm].push({artist, title})
  } else {
    info[bpm] = [{artist, title}]
  }

  return info
}, {})

export const supportedBpms = Object.keys(bpmInfo).map(bpm => Number(bpm))

export const beatLength = bpm => 60 / bpm
