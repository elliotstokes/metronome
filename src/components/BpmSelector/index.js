import React from 'react'
import { supportedBpms } from '../../config/bpm'
import './style.css'

const BpmVizualizer = ({selectedBpm, onSelectBpm}) => (
  <div className={'bpm-selector'}>
    {supportedBpms.map(bpm => (
    <div key={bpm}>
      <button className={`bpm-button${bpm === selectedBpm ? ' bpm-selected' : ''}`} onClick={()=> onSelectBpm(bpm) }>{`${bpm} BPM`}</button>
    </div>))}
  </div>
)

export default BpmVizualizer