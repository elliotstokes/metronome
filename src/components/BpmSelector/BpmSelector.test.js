import React from 'react'
import { create } from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'
import { supportedBpms } from '../../config/bpm'
import BpmSelector from '.'

test('renders correctly', () => {
  const component = create(<BpmSelector selectedBpm={supportedBpms[0]} onSelectBpm={jest.fn()} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('selecting a bpm calls the handler', () => {
  const onSelectBpmMock = jest.fn()
  const component = render(<BpmSelector selectedBpm={supportedBpms[0]} onSelectBpm={onSelectBpmMock} />)
  fireEvent.click(component.getByText('138 BPM'))
  expect(onSelectBpmMock).toBeCalledWith(138)
})