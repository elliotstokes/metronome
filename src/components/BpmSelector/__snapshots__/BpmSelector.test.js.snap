// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bpm-selector"
>
  <div>
    <button
      className="bpm-button bpm-selected"
      onClick={[Function]}
    >
      72 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      74 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      82 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      84 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      87 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      92 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      96 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      128 BPM
    </button>
  </div>
  <div>
    <button
      className="bpm-button"
      onClick={[Function]}
    >
      138 BPM
    </button>
  </div>
</div>
`;
